warehouse/utils/project.py

```diff
@@ -92,6 +98,64 @@ def prohibit_and_remove_project(
         remove_project(project, request, flash=flash)
 
 
+def quarantine_project(project: Project, request, flash=True) -> None:
+    """
+    Quarantine a project. Reversible action.
+    """
+    project.lifecycle_status = LifecycleStatus.QuarantineEnter
+    project.lifecycle_status_note = f"Quarantined by {request.user.username}."
+
+    project.record_event(
+        tag=EventTag.Project.ProjectQuarantineEnter,
+        request=request,
+        additional={"submitted_by": request.user.username},
+    )
```

Comment:
Nitpick: this tag might need another branch in the project security events HTML, to pretty-print it rather than dump it as a raw event 🙂

Reply:
But yes, absolutely - as part of the user-facing work that should follow this PR.